Adaptations for LVDS and PD23

s-hemer opened the pull request:
Finally

s-hemer committed:
README.md: Replace ftp download

Yocto recipe already replaced it.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
Add gitignore for downloaded image data

Add the downloaded tarball as well as the unpacked subdirectory to
gitignore.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
camvm016: Adopted video device to the correct one for kirkstone

While at it, use the 'videodev' variable and do the same for camera dev.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
aidemo: Fixed typo in printout

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
aidemo: Adopt frame sizes for LVDS

Also maximize window, reduce spacings and remove main label to fit the
smaller size constraints on the Pollux kit LVDS panel.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
aidemo: Fix spelling

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

smk-embedded commented:
oh! Danke für den PR! Ist das der Stand, der aktuell beim Dirk auf der HW läuft? Damit wären die Sourcen dann her gezaubert. Problem ist, dass wir auch die GPL für die Messedemos einhalten müssen. Also, bitte auch für Messedemos, müssen die Sourcen irgendwo abgelegt sein, wenn wir die präsentieren.

s-hemer commented:
Ja, das ist so in etwas das, was da läuft. @NBendisch und ich hatten das begonnen zusammen mit der Anpassung an PD23 (neue Demo mit neuerem SoM, nicht Dirks Variante) nachzubauen. Genau aus dem Grund, damit wir das ordentlich auch als Sourcen haben, haben wir das von Anfang an hier bzw in https://github.com/phytec/meta-celebrity-face-match/tree/WIP/n.bendisch%40phytec.de/update_PD23.1.0 ordentlich gemacht (mmh in phy2octo gibt es kein WIP, @NBendisch was ist da los? :wink: ). Nur der Feinschliff und der PR haben etwas auf sich warten lassen.

ACHTUNG: das Repo ist public...

s-hemer committed:
aidemo: Add a note on weston.ini edit for switch to LVDS

The SCREEN constant just switches the appearance of the demo but does
not automatically enable the LVDS-display.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer commented:
Grad gesehen, dass wir in meta-celebrity-face-match angefangen hatten, den default auf LVDS umzustellen, aber das hat so irgendwie nicht funktioniert. Deswegen hab ich das hier auch wieder rausgenommen. Vermutlich sollten wir da entscheiden, wie wir das machen wollen.

sriedmueller commented:
> Grad gesehen, dass wir in meta-celebrity-face-match angefangen hatten, den default auf LVDS umzustellen, aber das hat so irgendwie nicht funktioniert. Deswegen hab ich das hier auch wieder rausgenommen. Vermutlich sollten wir da entscheiden, wie wir das machen wollen.

Wenn ich das richtig sehe wird das KI Kit ohne Display ausgeliefert. Ich denke deshalb ist da HDMI als default eingestellt. Dann kann ein Kunde einfach nur ein HDMI Display dranhängen. Wenn wir jetzt LVDS als default einstellen, dann muss der Kunde das erst umstellen. Bin mir nicht sicher ob das so gewollt ist.

s-hemer commented:
Jep, deswegen ist es hier ja auch so gewählt. Dann muss es im meta-layer noch angepasst werden, zusammen mit der korrekten Revision des Merges.

s-hemer committed:
aidemo: Add option for fullscreen

LVDS demo usually runs fullscreen

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

sriedmueller commented on aidemo.py:
```diff
@@ -232,11 +244,13 @@ def setup_layout(self):
         content_box.pack_start(picture_box, True, True, 0)
         main_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=10)
-        main_box.pack_start(self.main_label, True, True, 0)
+        if SCREEN == 'HDMI' | FULLSCREEN:
```
```suggestion
        if SCREEN == 'HDMI' or FULLSCREEN:
```
`|` is a bitwise or which does not work with `str` and `bool`.
Alternatively you could use:
```suggestion
        if (SCREEN == 'HDMI') | FULLSCREEN:
```
Although the first one looks more appealing to me.

s-hemer replied:
Most appealing would be `if (SCREEN == 'HDMI' || FULLSCREEN) {...`
